src/api/app/controllers/status_project_controller.rb

```diff
@@ -1,41 +1,68 @@
 class StatusProjectController < ApplicationController
+  before_action :initialize_caches, only: [:show]
+
   def show
     dbproj = Project.get_by_name(params[:project])
     @packages = ProjectStatus::Calculator.new(dbproj).calc_status
-    find_relationships_for_packages(@packages)
+    find_relationships_for_packages
   end
 
   private
 
+  def initialize_caches
+    @package_hash = {}
+  end
+
   def role_from_cache(role_id)
```

Comment:
There is an `ActiveRecord` query cache. No need to implement a cache on top of it.

Reply:
@hennevogel I think merging oder don't merging this PR, the caching will be there anyway :)

Reply:
> @hennevogel 10 hours ago
> There is an ActiveRecord query cache. No need to implement a cache on top of it.

This sounds [familiar](https://github.com/openSUSE/open-build-service/pull/5765#issuecomment-417647592).

@vpereira And if this is really needed for performance reasons I would check if we could reuse the role caching we have in the [model](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/role.rb#L46).

Reply:
@bgeuken the code exists already, I just refactored it.  I did not implement the caching there. If you say that I can remove it, I do it..  regarding the usage of  ```Rails.cache.fetch``` vs memoization, I think both are valid, and we have probably both in our code base, Rails had even it's own ``` ActiveSupport::Memoizable``` but then Jose Valim, decided that the "ruby way" was good enough https://github.com/rails/rails/commit/36253916b0b788d6ded56669d37c96ed05c92c5c 

TL;DR:
You can just tell me, please remove that, and I do, I have no strong opinion on that. Otherwise I will keep it as it is (or will research more if needed), either way, IMO it's unrelated with the PR.

Reply:
> This sounds familiar.
> 
> @vpereira And if this is really needed for performance reasons I would check if we could reuse the role caching we have in the model.

@bgeuken I posted it [here](https://github.com/openSUSE/open-build-service/pull/5805/files#r215916132)

Reply:
Well, you are refactoring the code around these caching methods as well as the caching code. It's the second time we dance around this code now. I think it's worth to check if we even need this code before refactoring it (and if we cache the right thing when there is some doubt).

> vpereira 3 days ago
> @bgeuken the code exists already, I just refactored it. I did not implement the caching there. If you say that I can remove it, I do it..

You don't need me to answer that question:-)

Reply:
> Well, you are refactoring the code around these caching methods as well as the caching code. It's the second time we dance around this code now.

![lambada](https://user-images.githubusercontent.com/37418/45300504-d0ad7b80-b50e-11e8-81d5-021700d2212c.gif)

Reply:
I'm the last guy @bgeuken